assets/js/base/components/block-error-boundary/index.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import BlockError from './block-error';
+import './style.scss';
+
+interface DerivedStateReturn {
+	errorMessage: JSX.Element | string;
+	hasError: boolean;
+}
+
+interface Error {
+	status: string;
+	statusText: string;
+	message: string;
+}
+
+type RenderError = ( props: RenderErrorProps ) => React.ReactNode | null;
+
+interface RenderErrorProps {
+	errorMessage: React.ReactNode;
+}
+
+interface BlockErrorBoundaryProps {
```

Comment:
This seems like it has a lot in common with `BlockErrorProps`, what about extending that?

Reply:
You're right here and I realised that we are defining a lot of types in the index.tsx file so I've moved them all into a `types.ts` file.